### PR TITLE
Add PG option in contact action

### PR DESCRIPTION
**Pull request by dewenyushu**
Refs #23802

### CODE REVIEW

File: framework/doc/content/syntax/Constraints/Petrov-Galerkin_approach_for_Lagrange_multipliers.md [hunk 1]
```diff
@@ -21,10 +21,24 @@ where $\lambda$ and $\delta \lambda$ are the Lagrange multiplier variable and it
 
 The Petrov-Galerkin method is implemented in MOOSE by overwriting the Lagrange multiplier's (dual) test functions using that of a standard Lagrange multiplier variable. This functionality is enabled for all mortar-based constraints and the weighted-gap user object.
 
-!alert note
-This functionality currently only works for the `Constraints` system. Adding it to the `ContactAction` remains an ongoing process.
+To successfully use this approach in `ContactAction`, one just need to set `use_petrov_galerkin = true` in the `Contact` block. For example:
```
Comment (lindsayad) [inline framework/doc/content/syntax/Constraints/Petrov-Galerkin_approach_for_Lagrange_multipliers.md:24]:
```suggestion
To successfully use this approach in the contact action, one just needs to set `use_petrov_galerkin = true` in the `Contact` block. For example:
```

File: modules/contact/src/actions/ContactAction.C [hunk 2]
```diff
@@ -200,6 +200,15 @@ ContactAction::validParams()
...
       "The tag names for extra vectors that residual data should be saved into");
+  params.addParam<bool>(
+      "use_petrov_galerkin",
+      false,
+      "Whether to use the Petrov-Galerkin approach for the mortar-based constraints. If set to "
+      "true, we use the standard basis as the test function and dual basis as "
+      "the shape function for the interpolation of the Lagrange multiplier variable.");
+  params.addCoupledVar("aux_lm",
+                       "Auxiliary Lagrange multiplier variable that is utilized together with the "
+                       "Petrov-Galerkin approach.");
```
Comment (lindsayad) [inline modules/contact/src/actions/ContactAction.C:211]: It doesn't seem like you're using this parameter?

**Review by lindsayad — CHANGES_REQUESTED** [review]
Would it be possible to do some symlinking to results from a `Constraints` based test to prove that the action syntax and `Constraints` syntax produce the same result?

File: modules/contact/src/actions/ContactAction.C (hunk 2 above)
Comment (dewenyushu) [reply modules/contact/src/actions/ContactAction.C:211]: That's true! Nice catch! 

It was in my original version which should be removed after switching to automatically adding an aux LM variable

**Comment by dewenyushu** [comment]
> 



> Would it be possible to do some symlinking to results from a `Constraints` based test to prove that the action syntax and `Constraints` syntax produce the same result?

Thanks for the suggestion! I did try to use symlink to avoid creating more gold files, but quickly quit since some variable names are not consistent between the action and non-action versions (for Exodiff). Let me have both versions to use CSVdiff to avoid this issue.

**Review by lindsayad — APPROVED** [review]

**Pull request merged**